src/Libraries/Nop.Services/Localization/LocalizationService.cs

```diff
@@ -454,16 +456,11 @@ public virtual void ImportResourcesFromXml(Language language, StreamReader xmlSt
                 else
                 {
                     var lsr = new LocaleStringResource { LanguageId = language.Id, ResourceName = name, ResourceValue = value };
-                    if (lrsToInsertList.ContainsKey(name))
-                        lrsToInsertList[name] = lsr;
-                    else
-                        lrsToInsertList.Add(name, lsr);
+                    lrsToInsertList[name] = lsr;
                 }
             }
 
-            foreach (var lrsToUpdate in lrsToUpdateList) 
-                _lsrRepository.Update(lrsToUpdate);
-
+            _lsrRepository.Update(lrsToUpdateList);
             _lsrRepository.Insert(lrsToInsertList.Values);

```

Comment:
Hi @roman-gherman. Thank you for your contribution, I looked at your solution, everything is fine, but lrsToInsertList can now become really a List and not a Dictionary

Reply:
Hi @skoshelev, 
This is an Add or Update: lrsToInsertList[name] = lsr; 
This was done to avoid doing lrsToInsertList.ContainsKey(name).
 Complexity for the dictionary is O(1), which makes it faster.

If I convert lrsToInsertList to a List, then I will have to do a check for:
```
var item = lrsToInsertList.FirstOrDefault(lsr => lsr.ResourceName == name)
if(item  != null)
item = new LocaleStringResource....;
else
lrsToInsertList.Add(...)

```
Which makes it more complex and slow

Reply:
Yes, you may be right, I was mistaken with the opportunity to return to the List. Thanks again for your contribution.